### PR TITLE
Login

### PR DESCRIPTION
**Pull request by rutu0312**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -14,11 +12,10 @@ function App() {
    <BrowserRouter>
    <Routes>
     <Route path='/login' element={<Login/>}></Route>
+    <Route path='/homepage' element={<SideBar/>}></Route>
```
Comment (pooja-vipulbhai-darji) [inline src/App.js:15]: format all files.

File: src/Pages/Login/login.jsx [hunk 2]
```diff
@@ -1,54 +1,129 @@
...
+        },
+
+        validationSchema: basicSchema,
+        onSubmit: async (values) => {
+            console.log(values);
+            await fetch(`${baseURL}login`, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Access-Control-Allow-Origin': '*'
```
Comment (pooja-vipulbhai-darji) [inline src/Pages/Login/login.jsx:29]: please write the API functions in different file

File: src/Pages/Login/login.jsx [hunk 3]
```diff
@@ -1,54 +1,129 @@
...
+                },
+                body: JSON.stringify(values)
+            })
+                .then(response => response.json())
+                .then(res => {
+                    if (res.status === 'success') {
```
Comment (pooja-vipulbhai-darji) [inline src/Pages/Login/login.jsx:35]: for the status use HTTP Status Code package

File: src/Pages/Login/login.jsx [hunk 4]
```diff
@@ -1,54 +1,129 @@
...
+
+    const formik = useFormik({
+        initialValues: {
+            email: "",
+            password: ""
+        },
+
+        validationSchema: basicSchema,
+        onSubmit: async (values) => {
+            console.log(values);
```
Comment (pooja-vipulbhai-darji) [inline src/Pages/Login/login.jsx:24]: remove the consoles

File: src/Pages/Login/login.jsx [hunk 5]
```diff
@@ -1,54 +1,129 @@
...
+                    'Content-Type': 'application/json',
+                    'Access-Control-Allow-Origin': '*'
+                },
+                body: JSON.stringify(values)
+            })
+                .then(response => response.json())
+                .then(res => {
+                    if (res.status === 'success') {
+                        toast.success(res.message)
+                        setTimeout(()=> {
```
Comment (pooja-vipulbhai-darji) [inline src/Pages/Login/login.jsx:37]: why you put the setTimeout

File: src/Pages/Login/login.jsx [hunk 6]
```diff
@@ -1,54 +1,129 @@
...
+            email: "",
+            password: ""
+        },
+
+        validationSchema: basicSchema,
+        onSubmit: async (values) => {
+            console.log(values);
+            await fetch(`${baseURL}login`, {
+                method: 'POST',
+                headers: {
```
Comment (pooja-vipulbhai-darji) [inline src/Pages/Login/login.jsx:27]: why you have not add the loader?

File: src/Pages/Login/login.jsx [hunk 7]
```diff
@@ -1,54 +1,129 @@
...
                                                 </div>
-                                            </div>
-                                            <div className="col-md-6">
-                                                <div className="form-check form-switch">
-                                                    <input className="form-check-input" type="checkbox" id="flexSwitchCheckChecked" />
-                                                        <label className="form-check-label" for="flexSwitchCheckChecked">Remember Me</label>
+                                                <div className="col-12">
+                                                    <label for="inputChoosePassword" className="form-label">Enter Password</label>
+                                                    <div className="col-12" id="show_hide_password">
+
```
Comment (pooja-vipulbhai-darji) [inline src/Pages/Login/login.jsx:84]: remove the unnecessary space

File: src/Pages/Login/login.jsx [hunk 8]
```diff
@@ -1,54 +1,129 @@
...
     return (
-        <div className="section-authentication-signin d-flex align-items-center justify-content-center my-5 my-lg-4">
-            <div className="container-fluid">
-                <div className="row row-cols-1 row-cols-lg-2 row-cols-xl-3">
-                    <div className="col mx-auto">
-                        <div className="card mt-5 mt-lg-0">
-                            <div className="card-body">
-                                <div className="border p-4 rounded">
-                                    <div className="text-center">
-                                        <h3 className="pb-4 m-3">Sign in</h3>
```
Comment (pooja-vipulbhai-darji) [inline src/Pages/Login/login.jsx:14]: make a constant

File: src/Pages/Login/login.jsx [hunk 9]
```diff
@@ -1,54 +1,129 @@
...
+                                                    <Link to='/forgotpassword'>
+                                                    Forgot Password ?
+                                                    </Link>
+                                                    	
                                                 </div>
-                                            </div>
-                                        </form>
+                                                <div className="col-12">
+                                                    <div className="d-grid">
+                                                        {/* <BaseButton label='Submit' color="success" /> */}
```
Comment (pooja-vipulbhai-darji) [inline src/Pages/Login/login.jsx:112]: remove the commented code

File: src/Schema/index.jsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    email:yup
+    .string().
+    email("Please enter a valid email")
+    .required("Required"),
+
+    password: yup
+    .string()
+    .min(5)
+    .matches(passwordRules,{message:"Invalid Format"})
+    .required("Required"),
```
Comment (pooja-vipulbhai-darji) [inline src/Schema/index.jsx:16]: Message Should be Email is required.
(is required.) is in common function and use everywhere.

File: src/Pages/Login/login.jsx (hunk 5 above)
Comment (rutu0312) [reply src/Pages/Login/login.jsx:37]: because toast button can be shown

File: src/Pages/Login/login.jsx [hunk 11]
```diff
@@ -1,56 +1,117 @@
...
+                            <div className="card mt-5 mt-lg-0">
+                                <div className="card-body">
+                                    <div className="border p-4 rounded">
+                                        <div className="text-center">
+                                            <h3 className="pb-4 m-3">{SignIn}</h3>
+                                        </div>
+                                        <div className="form-body">
+                                            <form className="row g-3" onSubmit={formik.handleSubmit}>
+                                                <div className="row-12">
+                                                    <label for="inputEmailAddress" className="form-label">Email Address</label>
```
Comment (pooja-vipulbhai-darji) [inline src/Pages/Login/login.jsx:56]: Still i am not able to show the constant

File: src/constant/index.js [hunk 12]
```diff
@@ -0,0 +1,10 @@
+export const SignIn = 'Sign In'
+export const ForgotPassword = 'Forgot Password ?'
+export const LoginHeader = {
+    'Content-Type': 'application/json',
+    'Access-Control-Allow-Origin': '*'
+}
+
+export const required = ' is Required'
```
Comment (pooja-vipulbhai-darji) [inline src/constant/index.js:8]: Remove this

File: src/enums/index.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
+import { StatusCodes } from "http-status-codes"
+
+export const statusCode = {
+    SUCCESS: "success" || StatusCodes.OK || StatusCodes.CREATED || StatusCodes.NO_CONTENT,
+    ERROR: "error" || StatusCodes.BAD_REQUEST || StatusCodes.UNAUTHORIZED || StatusCodes.FORBIDDEN || StatusCodes.NOT_FOUND || StatusCodes.INTERNAL_SERVER_ERROR || StatusCodes.BAD_GATEWAY || StatusCodes.SERVICE_UNAVAILABLE || StatusCodes.GATEWAY_TIMEOUT
+}
```
Comment (pooja-vipulbhai-darji) [inline src/enums/index.js:6]: you need to make a different different file

**Pull request merged**